### PR TITLE
Fixes edge-case for rsync detection

### PR DESCRIPTION
**Pull request by msheiny**
The current rsync detection logic afflicted other test roles that depended on rsyncin' the files
around as opposed to using git_pull. Seemed to be kind of inconsistent with failures - I'm
surprised this wasnt caught before to be honest.

### CODE REVIEW

File: tasks/pull_sitecode.yml [hunk 1]
```diff
@@ -56,7 +56,7 @@
     rsync_opts: "{{ django_stack_rsync_opts }}"
   when:
     - not django_stack_git_repo
-    - django_stack_deploy_src
+    - "not django_stack_deploy_src == ''"
```
Comment (conorsch) [inline tasks/pull_sitecode.yml:59]: `"django_stack_deploy_src != ''"` reads better to me, but I can deal.

**Review by conorsch — APPROVED** [review]
Concise changes, default role behavior preserved, but override capability added. :+1:
